WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -179,7 +182,8 @@ public static Statement ZeroProofStatement(GroupElement ma)
 
 		public static Statement RangeProofStatement(GroupElement ma, IEnumerable<GroupElement> bitCommitments)
```

Comment:
with prev comment, the following signature would make more sense:

```suggestion
		public static Statement RangeProofStatement(GroupElement ma, GroupElementVector bitCommitments)
```

at which point the `.ToArray()` stuff can be removed as well, which can also clean up the equation construction a bit

Reply:
I am not so sure about this. The general rule is to receive the more generic type.

Reply:
generally I'd agree but I think in this case it makes more sense because there's not much value in keeping it generic, we need to refer to the elements multiple times by their index